pinot-common/src/main/java/org/apache/pinot/pql/parsers/PinotQuery2BrokerRequestConverter.java

```diff
@@ -0,0 +1,362 @@
+        selectionSort.setIsAsc(false);
+      }
+      selectionSort.setColumn(orderByExpr.getFunctionCall().getOperands().get(0).getIdentifier().getName());
+      sortSequenceList.add(selectionSort);
+    }
+    if (!sortSequenceList.isEmpty()) {
+      brokerRequest.getSelections().setSelectionSortSequence(sortSequenceList);
+    }
+  }
+
+  private void handleGroupBy(PinotQuery pinotQuery, BrokerRequest brokerRequest) {
+    List<Expression> groupByList = pinotQuery.getGroupByList();
+    if (groupByList != null && groupByList.size() > 0) {
+      GroupBy groupBy = new GroupBy();
+      for (Expression expression : groupByList) {
+        String expressionStr = standardizeExpression(expression, true);
+        groupBy.addToExpressions(expressionStr);
+      }
+      groupBy.setTopN(pinotQuery.getLimit());
+      brokerRequest.setGroupBy(groupBy);
+    }
+  }
+
+  private void handleSelectList(PinotQuery pinotQuery, BrokerRequest brokerRequest) {
+    Selection selection = null;
+    List<AggregationInfo> aggregationInfoList = null;
+    for (Expression expression : pinotQuery.getSelectList()) {
+      ExpressionType type = expression.getType();
+      switch (type) {
+        case LITERAL:
+          if (selection == null) {
+            selection = new Selection();
+          }
+          selection.addToSelectionColumns(expression.getLiteral().getStringValue());
+          break;
+        case IDENTIFIER:
+          if (selection == null) {
+            selection = new Selection();
+          }
+          selection.addToSelectionColumns(expression.getIdentifier().getName());
+          break;
+        case FUNCTION:
+          AggregationInfo aggInfo = buildAggregationInfo(expression.getFunctionCall());
+          if (aggregationInfoList == null) {
+            aggregationInfoList = new ArrayList<>();
+          }
+          aggregationInfoList.add(aggInfo);
+          break;
+      }
+    }
+
+    if (selection != null) {
+      if (pinotQuery.isSetOffset()) {
+        selection.setOffset(pinotQuery.getOffset());
+      }
+      if (pinotQuery.isSetLimit()) {
+        selection.setSize(pinotQuery.getLimit());
+      }
+      brokerRequest.setSelections(selection);
+    }
+
+    if (aggregationInfoList != null && aggregationInfoList.size() > 0) {
+      brokerRequest.setAggregationsInfo(aggregationInfoList);
+    }
+  }
+
+  private void handleFilter(PinotQuery pinotQuery, BrokerRequest brokerRequest) {
+    Expression filterExpression = pinotQuery.getFilterExpression();
+
+    //Handle filter
+    if (filterExpression != null) {
+      FilterQuery filterQuery;
+      FilterQueryMap filterSubQueryMap = new FilterQueryMap();
+      filterQuery = traverseFilterExpression(filterExpression, filterSubQueryMap);
+      brokerRequest.setFilterQuery(filterQuery);
+      brokerRequest.setFilterSubQueryMap(filterSubQueryMap);
+    }
+  }
+
+  private String standardizeExpression(Expression expression, boolean treatLiteralAsIdentifier) {
```

Comment:
IIRC, this is probably inspired from some existing code? The name standardize is quite generic, but I don't have a better name either (may be sanitze??). Would be nice to have some javadoc with description and examples.